test/src/androidTest/kotlin/io/realm/shared/RealmListTests.kt

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.shared
+
+import io.realm.MutableRealm
+import io.realm.Realm
+import io.realm.RealmConfiguration
+import io.realm.RealmList
+import io.realm.RealmObject
+import io.realm.util.PlatformUtils
+import io.realm.util.TypeDescriptor
+import test.list.RealmListContainer
+import kotlin.reflect.KClassifier
+import kotlin.reflect.KMutableProperty1
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertTrue
+
+class RealmListTests {
+
+    private val descriptors = TypeDescriptor.allListFieldTypes
+
+    private lateinit var tmpDir: String
+    private lateinit var realm: Realm
+
+    @BeforeTest
+    fun setup() {
+        tmpDir = PlatformUtils.createTempDir()
+        val configuration = RealmConfiguration(
+            path = "$tmpDir/default.realm",
+            schema = setOf(RealmListContainer::class)
+        )
+        realm = Realm.open(configuration)
+    }
+
+    @AfterTest
+    fun tearDown() {
+        if (!realm.isClosed()) {
+            realm.close()
+        }
+        PlatformUtils.deleteTempDir(tmpDir)
+    }
+
+    @Test
+    fun add() {
+        for (tester in unmanagedTesters) {
+            tester.add()
+        }
+        for (tester in managedTesters) {
+            tester.add()
+        }
+    }
+
+    @Test
+    fun addWithIndex() {
+        for (tester in unmanagedTesters) {
+            tester.addWithIndex()
+        }
+        for (tester in managedTesters) {
+            tester.addWithIndex()
+        }
+    }
+
```

Comment:
Is there a possibility to test the order `insertAt` is honored on a managed list?

Reply:
Here I iterate over `typeSafetyManager.getInitialDataSet()`'s elements (which is the original dataset I use to populate the `list`) using indices and check that they are in the rightful position after inserting them.
```
                typeSafetyManager.getInitialDataSet()
                    .forEachIndexed { index, e ->
                        assertEquals(index, list.size)
                        list.add(0, copyToRealmIfNeeded(e))
                        assertEquals(index + 1, list.size)
                    }
```
Is this what you meant?

Reply:
Oh, you're right. The element equality assertion is missing. Sorry, it's coming in the next PR 😅 This one was just to set framework in place.